packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -650,21 +637,18 @@ export class SDKClient {
 
       throw sdkClientError;
     } finally {
-      /**
-       * @note Capturing the charged transaction fees at the end of the flow ensures these fees are eventually
-       *       captured in the metrics and rate limiter class, even if SDK transactions fail at any point.
-       */
+      // emitting an EXECUTE_QUERY event to kick off capturing metrics process asynchronously
       if (queryCost && queryCost !== 0) {
-        this.addExpenseAndCaptureMetrics(
-          `TransactionExecution`,
-          query.paymentTransactionId!,
-          queryType,
+        this.eventEmitter.emit(constants.EVENTS.EXECUTE_QUERY, {
+          executionType: `TransactionExecution`,
+          transactionId: query.paymentTransactionId?.toString()!,
```

Comment:
```suggestion
          transactionId: query.paymentTransactionId!.toString(),
```

Reply:
Even better, wrap this with `if (query.paymentTransactionId) { ... }` to make sure that we are not going to run into this case. And then inside of the `else` block, we can log a warning/error for situations when the `paymentTransactionId` is missing from the query

Reply:
The `paymentTransactionId()` is not a critical value used for looking up transactions or any other processing but rather for logging purposes in addExpenseAndCaptureMetrics(). A query may or may not have a `paymentTransactionId`, but the `queryCost` is always calculated regardless. Therefore, the metrics capturing logic should depend on `queryCost` rather than `paymentTransactionId`.

Reply:
Okay, then let’s remove the `!` as it would throw an error if the paymentTransactionId is undefined or null:
```suggestion
          transactionId: query.paymentTransactionId?.toString(),
```

Reply:
Is the operator account currently paying for all queries? 

If not, are we doing any checks that the `payerAccountId === operatorAccountId` before we add the query expenses to the `HbarLimiter`?

Reply:
Yes, by default, the operator account pays for all queries. According to [this source](https://docs.hedera.com/hedera/sdks-and-apis/sdks/client#id-2.-define-the-operator-account-id-and-private-key):

> The operator is the account that will, by default, pay the transaction fee for transactions and queries built with this client. The operator account ID is used to generate the default transaction ID for all transactions executed with this client. The operator private key is used to sign all transactions executed by this client.